### PR TITLE
chore: pull CRS from `common` into `acvm-backend-barretenberg`

### PR DESCRIPTION
**Pull request by TomAFrench**
This pulls across the CRS downloading logic into `acvm-backend-barretenberg`.

### CODE REVIEW

File: acvm_backend_barretenberg/Cargo.toml [hunk 1]
```diff
@@ -28,15 +35,18 @@ tempfile = "*"
 
 [features]
 default = ["native"]
+
+std = ["dep:reqwest", "dep:tokio", "dep:futures-util", "dep:dirs", "dep:indicatif"]
```
Comment (phated) [inline acvm_backend_barretenberg/Cargo.toml:39]: I think we just drop this feature. These impls are going to be behind native/wasm/js feature flags. No reason to keep this.

File: acvm_backend_barretenberg/src/composer.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use common::barretenberg_structures::*;
-use common::crs::{CRS, G2};
 
+use crate::crs::{CRS, G2};
```
Comment (phated) [inline acvm_backend_barretenberg/src/composer.rs:3]: G2 is going away with reference string changes

**Review by phated — CHANGES_REQUESTED** [review]

File: acvm_backend_barretenberg/src/composer.rs (hunk 2 above)
Comment (TomAFrench) [reply acvm_backend_barretenberg/src/composer.rs:3]: I think let's keep it for now. If changes are required for the new interface changes then we can adjust this at the same time.

File: acvm_backend_barretenberg/Cargo.toml (hunk 1 above)
Comment (TomAFrench) [reply acvm_backend_barretenberg/Cargo.toml:39]: Would you prefer to inline all of these dependencies into `native` and `wasm`? That seems noisy to me and allows for  them to get out of sync.

Comment (phated) [reply acvm_backend_barretenberg/Cargo.toml:39]: Yeah, I think we should. I'd rather do work to keep the features in sync than to have confusing combinations of features.

File: acvm_backend_barretenberg/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,14 @@ rust-version.workspace = true
 
 [dependencies]
 blake2 = "0.9.1"
-common = { path = "../common", default-features = false }
+common = { path = "../common" }
+
+# std
```
Comment (phated) [inline acvm_backend_barretenberg/Cargo.toml:15]:
```suggestion
```
nit

File: acvm_backend_barretenberg/Cargo.toml [hunk 4]
```diff
@@ -28,20 +35,29 @@ tempfile = "*"
 
 [features]
 default = ["native"]
+
```
Comment (phated) [inline acvm_backend_barretenberg/Cargo.toml:38]:
```suggestion
```
nit

**Review by phated — APPROVED** [review]
lgtm, can you cleanup the "behaviour" cspell? Silly american english

**Pull request merged**